drafts/AENS.md

```diff
@@ -0,0 +1,414 @@
+# Aeternity Naming System
+
+
+## Abstract
+
+This document describes the first draft of the Aeternity Naming System,
+short AENS.
+
+
+## Motivation
+
+Entities native to most blockchain systems are addressed or identified
+by their hashes, which are generally `n` bit numbers displayed in a hex
+or base64/base58 notation, all of which are somewhat unfriendly to the
+humans trying to use these systems.
+
+
+## Background
+
+A solution to a similar problem is the Domain Name System, or DNS, which
+should be familiar to most people, providing human readable names for IP
+addresses. The issue with DNS is its centralization and it was conjectured
+by [Zooko](https://en.wikipedia.org/wiki/Zooko%27s_triangle) that a 
+decentralized system can't have human-meaningful names while still being
+secure. This conjecture has been proven to be false by a number of systems,
+one example being [NameCoin](https://www.namecoin.org), which was debuted
+in 2011. It allowed users to associate names with a number of key-value
+pairs. The NameCoin network still exists but sees barely any usage.
+
+A more recent effort is ENS, the Ethereum Name Service, which enables users 
+to claim names via a set of smart contracts. ENS is currently running a two
+year trial deployment, during which users can claim name with a `.eth` 
+postfix, e.g. `mywallet.eth`. ENS currently allows users to associate a single
+32 byte array of data with a node, typically used for a hash.
+
+
+## Overview
+
+Names are part of a namespace, which works just like DNS, e.g. `mywallet.aet`
+or `привет.aet` are part of the same `.aet` namespace.
+
+
+## Governance
+
+This first draft of the AENS is not going to have any governance mechanism but
+will only allow registrations under a single namespace: `.aet`.
+We will also not include mechanisms similar to the `sunrise` and `landrush`
+periods, which are common for DNS, where registration is restricted to small 
+select groups and trademark holders in order to avoid name squatting, before
+the general public can start claiming names.
+
+
+## Mechanisms
+
+It is unclear what a good mechanism for a naming system would look
+like. If we imagine two actors both being interested in the same name,
+what would a »fair« solution be to resolve this?
+
+Fees are the main mechanism to discourage spam and squatting.
+
+***Options:***
+
+1. one time fee, which can be claimed by miner
+2. one time fee which gets burned
+3. deposit which can be reclaimed after the name is released
+4. no additional fee other than the standard transaction fee
+
```

Comment:
3. is quite weak to prevent squatting and spam, 2. seems to be the safest option to prevent fraud for now and the most useful option might be to collect them and use the governance system to invest them into the Eco-system, for example: improve the naming system or other parts of the blockchain.

Reply:
Allocating them for the network was an idea I had as well but we don't have a mechanism for that just yet. I'll include it, though. It would definitely be nice if the network was able to fund itself.